File: frontend/src/mockData.ts
import { Dictionary, OrderSide, OrderStatus, TickerData, TradeData } from "./types";

const MOCK_TICKER_DATA_ENTRIES = 12;

function generateMockTickerData(numEntries: number): Dictionary<TickerData> {
  const mockStockData: Dictionary<TickerData> = {};

  for (let i = 0; i < numEntries; i++) {
    const symbol = generateRandomSymbol();
    mockStockData[symbol] = {
      name: `Company ${symbol}`,
      volumeTraded: generateRandomNumber(100, 1000),
      highTradePrice: generateRandomNumber(80, 120),
      lowTradePrice: generateRandomNumber(60, 90),
      vwap: generateRandomNumber(70, 110),
    };
  }

  return mockStockData;
}

function generateRandomSymbol(): string {
  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
  const randomCharacter = characters[Math.floor(Math.random() * characters.length)];
  const randomNumber = generateRandomNumber(100, 999);
  return `${randomCharacter}${randomNumber}`;
}

function generateRandomNumber(min: number, max: number): number {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

export const mockTickers = generateMockTickerData(MOCK_TICKER_DATA_ENTRIES);

export const mockTradeA = {
    order_id: "b64957f2-c60d-47af-83fe-e4b9e9fa288d",
    volume: 25,
    price: 100,
    side: OrderSide.B,
    date: new Date(),
    ticker: "AAPL",
    status: OrderStatus.SUCCESS,
};

export const mockTradeB = {
  order_id: "b64957f2-c60d-47af-83fe-e4b9e9fa288d",
  volume: 25,
  price: 100,
  side: OrderSide.B,
  date: new Date(),
  ticker: "MSFT",
  status: OrderStatus.SUCCESS,
};

export const generateRandomTrade = (): TradeData => {
  const randomTicker = Object.keys(mockTickers)[Math.floor(Math.random() * MOCK_TICKER_DATA_ENTRIES)];
  const randomSide = Math.random() > 0.5 ? OrderSide.B : OrderSide.O;
  const randomVolume = generateRandomNumber(1, 100);
  const randomPrice = generateRandomNumber(50, 150);

  return {
    order_id: generateRandomSymbol(),
    volume: randomVolume,
    price: randomPrice,
    side: randomSide,
    date: new Date(),
    ticker: randomTicker,
    status: OrderStatus.SUCCESS,
  };
}

File: api/mock_data.py
from datetime import datetime
import random

stock_tickers = ["AAPL", "MSFT", "AMZN", "GOOGL", "META", "TSLA", "JNJ", "JPM"]

class OrderSide:
    B = "B"
    O = "O"

class OrderStatus:
    SUCCESS = "SUCCESS"


def generate_mock_ticker_data(num_entries):
    mock_stock_data = {}

    for _ in range(num_entries):
        symbol = generate_random_symbol()
        mock_stock_data[symbol] = {
            "name": stock_tickers[_ % len(stock_tickers)],
            "volume_traded": generate_random_number(100, 1000),
            "high_trade_price": generate_random_number(80, 120),
            "low_trade_price": generate_random_number(60, 90),
            "vwap": generate_random_number(70, 110),
        }

    return mock_stock_data

def generate_random_symbol():
    characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
    random_character = random.choice(characters)
    random_number = generate_random_number(100, 999)
    return f"{random_character}{random_number}"

def generate_random_number(min_val, max_val):
    return random.randint(min_val, max_val)

def generate_random_trade():
    random_ticker = random.choice(stock_tickers)
    random_side = OrderSide.B if random.random() > 0.5 else OrderSide.O
    random_volume = generate_random_number(1, 100)
    random_price = generate_random_number(50, 150)

    return {
        "order_id": generate_random_symbol(),
        "volume": random_volume,
        "price": random_price,
        "side": random_side,
        "date": datetime.now(),
        "ticker": random_ticker,
        "status": OrderStatus.SUCCESS,
    }
...
mock_ticker_data = generate_mock_ticker_data(len(stock_tickers))
mock_trade_data = [generate_random_trade() for _ in range(8)]
in_memory_datastore = {
   "ticker_data": mock_ticker_data,
   "trade_data": mock_trade_data,
}

print(in_memory_datastore)